tests/test_search.py

```diff
@@ -99,6 +99,10 @@ def test_nested_tuple(self):
         self.client.search(['NOT', ('SUBJECT', 'topic',  'TO', 'some@email.com')])
         self.check_call([b'NOT', b'(SUBJECT', b'topic', b'TO', b'some@email.com)'])
 
+    def test_quote_empty_strings(self):
```

Comment:
I would either merge this test into `test_quoting` (line 63) or place it just under, as they are related.

By the way I am wondering why there is no tests to check the escape of backslashes, but this is not in the scope of your PR :)

Reply:
👍